src/wallet/coinselection.cpp

```diff
@@ -102,7 +102,6 @@ bool SelectCoinsBnB(std::vector<CInputCoin>& utxo_pool, const CAmount& target_va
             // explore any more UTXOs to avoid burning money like that.
             if (curr_waste <= best_waste) {
                 best_selection = curr_selection;
-                best_selection.resize(utxo_pool.size());
```

Comment:
I think you will need to also record `curr_selection_size` since copying `curr_selection` may also copy outputs that were selected in previous iterations that are not actually part of the current selection (i.e. selected previously but are at indexes past `curr_selection_size`).

Reply:
I think this is not necessary. If I understand it correctly, what's right from `curr_selection_size` will always be 0. The only place where the size goes back is in line 114-117, the loop stops when it finds `true` then sets it to `false`. So there is no way that any value right from `curr_selection_size` will ever be true

Reply:
Indeed, that is the case. Ignore my other comments then.